src/Microsoft.Framework.Runtime.Roslyn/RoslynCompilationFailure.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Framework.Runtime.Roslyn
+{
+    /// <summary>
+    /// An implementation of <see cref="ICompilationFailure"/> for Roslyn compilation.
+    /// </summary>
+    public class RoslynCompilationFailure : ICompilationFailure
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="RoslynCompilationFailure"/>.
+        /// </summary>
+        /// <param name="diagnostics">A sequence of <see cref="Diagnostic"/>s from Roslyn compilation.</param>
+        public RoslynCompilationFailure(IEnumerable<Diagnostic> diagnostics)
+        {
+            Messages = diagnostics.Select(diagnostic => new RoslynCompilationMessage(diagnostic));
+        }
+
+        /// <inheritdoc />
+        public IEnumerable<ICompilationMessage> Messages { get; }
+
+        /// <inheritdoc />
+        public string SourceFilePath { get; }
+
+        /// <inheritdoc />
+        public string SourceFileContent
```

Comment:
For Mvc, this could be content from an embedded file system since you can set `RazorViewEngineOptions.FileSystem`. Helios \ Diagnostics wouldn't be able to read it unless we pass the `IFileSystem` along. Consequently since Mvc already has the file content in memory, we assign the property value.